### ISSUE TITLE
NaN loss

### ISSUE THREAD

**Issue by nealchecka** [pankajmishra000/VT-ADL#4]
Hi Pankaj,

I'm following your README and attempting to train the anomaly detection for 'Hazelnut'. I've set all the training hyperparameters according to the paper (Adams LR = 0.0001, weight decay = 0.0001, batch size = 8, epochs = 400, No. of Gaussians =150, patch dim = 64). I noticed that after 298 epochs, the loss becomes NaN and the min loss is around 849. Is this the expected behavior? 
Any hints to prevent the NaN?

Thanks,
Neal

**Comment by pankajmishra000** [comment]
Follow the [link](https://github.com/pankajmishra000/VT-ADL/issues/1#issuecomment-882037806) and probably you will fix the issue- https://github.com/pankajmishra000/VT-ADL/issues/1#issuecomment-882037806

**Comment by nealchecka** [comment]
I did follow that link. I raised the number of Gaussian to 200, anything higher I run out of GPU memory. What value did you use to report the results in the paper? And what kind of loss did you end up achieving after 400 epochs?

**Comment by pankajmishra000** [comment]
I used 150 and other hyperparameters are also reported in the paper. The code committed here is not the exact code of the paper, as we tweaked some hyperparameters and some parts in GMM network to safeguard the sponsoring company's interest. I suggest you bring down the LR and it will work.

**Comment by pankajmishra000** [comment]
I am closing the issue with the current version
